Lib/_pyrepl/completing_reader.py

```diff
@@ -258,10 +258,15 @@ def after_command(self, cmd: Command) -> None:
     def calc_complete_screen(self) -> list[str]:
         screen = super().calc_complete_screen()
         if self.cmpltn_menu_visible:
-            ly = self.lxy[1]
+            # We display the completions menu below the current prompt
+            ly = self.lxy[1] + 1
```

Comment:
I am not sure, would you mind posting the relevant part of your patch? The problem is not with screen but screeninfo, which is used to calculate cursor position. Have you tried pressing the left or down arrow while the completion menu is displayed? Without the hack below the cursor would jump down into the completions menu in my case, but perhaps you solution works better.

Reply:
I think that this should be enough once #120042 lands.

```diff
diff --git a/Lib/_pyrepl/completing_reader.py b/Lib/_pyrepl/completing_reader.py
index c11d2dabdd2..db3c26c0627 100644
--- a/Lib/_pyrepl/completing_reader.py
+++ b/Lib/_pyrepl/completing_reader.py
@@ -258,10 +258,7 @@ def after_command(self, cmd: Command) -> None:
     def calc_complete_screen(self) -> list[str]:
         screen = super().calc_complete_screen()
         if self.cmpltn_menu_visible:
-            ly = self.lxy[1]
-            screen[ly:ly] = self.cmpltn_menu
-            self.screeninfo[ly:ly] = [(0, [])]*len(self.cmpltn_menu)
-            self.cxy = self.cxy[0], self.cxy[1] + len(self.cmpltn_menu)
+            screen.extend(self.cmpltn_menu)
         return screen
 
     def finish(self) -> None:
```

Reply:
I am going to have to look at that PR more closely, but just looking at this patch, it seems that the completions menu would always get appended at the very bottom of the screen right?

I thought about doing it this way as well, but I don't know if this will work well during multiline editing. You can imagine a corner case where you would be editing a large multiline code, and the completions menu will appear well below  the current cursor position or even outside the visible screen.
(I am currently AFK so cannot test this)

Reply:
> I am going to have to look at that PR more closely, but just looking at this patch, it seems that the completions menu would always get appended at the very bottom of the screen right?
I thought about doing it this way as well, but I don't know if this will work well during multiline editing. You can imagine a corner case where you would be editing a large multiline code, and the completions menu will appear well below the current cursor position or even outside the visible screen.

@lysnikolaou I have now tested your version and for large multiline editing it is indeed a problematic since the menu will always be displayed at the very bottom, even if you're editing a line at the very top of screen. For large enough editing context, the menu will not be visible at all (e.g. when editing a first line in a function with many lines. WDYT?

Reply:
Is this problem still relevant?

Reply:
To clarify, this PR has no problem afaik. I was pointing out issues with the suggestion above.